pkg/resource/route_table/hooks.go

```diff
@@ -62,36 +62,19 @@ func (rm *resourceManager) syncRoutes(
 		}
 	}
 
-	for _, desiredRoute := range desired.ko.Spec.Routes {
-		if (*desiredRoute).GatewayID != nil && *desiredRoute.GatewayID == LocalRouteGateway {
-			// no-op for default route
-			continue
-		}
-		if latestRoute := getMatchingRoute(desiredRoute, latest); latestRoute != nil {
-			delta := compareCreateRouteInput(desiredRoute, latestRoute)
-			if len(delta.Differences) > 0 {
-				// "update" route by deleting old route and adding the new route
-				toDelete = append(toDelete, latestRoute)
-				toAdd = append(toAdd, desiredRoute)
-			}
-		} else {
-			// a desired route is not in latest; therefore, create
-			toAdd = append(toAdd, desiredRoute)
-		}
-	}
-	if latest != nil {
-		for _, latestRoute := range latest.ko.Spec.Routes {
-			if (*latestRoute).GatewayID != nil && *latestRoute.GatewayID == LocalRouteGateway {
-				// no-op for default route
-				continue
-			}
-			if desiredRoute := getMatchingRoute(latestRoute, desired); desiredRoute == nil {
-				// latest has a route that is not desired; therefore, delete
-				toDelete = append(toDelete, latestRoute)
-			}
-		}
+	var toAdd, toDelete []*svcapitypes.CreateRouteInput
+
+	switch {
+	// If the route table is created all routes need to be added.
+	case delta == nil:
+		toAdd = removeLocalRoute(desired.ko.Spec.Routes)
+	// If there are changes to the routes in the delta ...
+	case delta.DifferentAt("Spec.Routes"):
+		toAdd, toDelete = filterDifferentRoutes(desired.ko.Spec.Routes, latest.ko.Spec.Routes)
+	default: // nothing to do
```

Comment:
why are we adding if delta is nil?
```suggestion
	if delta.DifferentAt("Spec.Routes") {
	    toAdd, toDelete = filterDifferentRoutes(desired.ko.Spec.Routes, latest.ko.Spec.Routes)
        }
```

Reply:
wow, good catch! I added `delta` to the signature, because I was using the differences for routes from it. Now that we recompute using `getRoutesDifference`, this is no longer necessary.